pilot/pkg/kube/inject/testdata/cronjob.yaml.injected

```diff
@@ -95,7 +95,7 @@ spec:
             - -x
             - ""
             - -b
-            - '*'
+            - ""
```

Comment:
This is a problem. When there is no containerPort listed, this fix will completely firewall the container, so may break apps that do not declare their list of ports.

This is what the spec says:
"portsContainerPort array patch type: merge patch merge key: containerPort | List of ports to expose from the container. Exposing a port here gives the system additional information about the network connections a container uses, but is primarily informational. Not specifying a port here DOES NOT prevent that port from being exposed. Any port which is listening on the default "0.0.0.0" address inside a container will be accessible from the network. Cannot be updated."

Reply:
I think with mtls we have to fail closed
(and document the behavior that you need to declare you ports to use istio injection)